src/Operation/Squash.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace loophp\collection\Operation;
+
+use Closure;
+use Generator;
+use Iterator;
+use Throwable;
+
+/**
+ * @psalm-template TKey
+ * @psalm-template TKey of array-key
+ * @psalm-template T
+ */
+final class Squash extends AbstractOperation
+{
+    /**
+     * @psalm-return Closure(bool): Closure(Iterator<TKey, T>): Closure(Iterator<TKey, T>): Iterator<TKey, T>
+     */
+    public function __invoke(): Closure
+    {
+        return
+            /**
+             * @psalm-return Closure(Iterator<TKey, T>): Closure(Iterator<TKey, T>): Iterator<TKey, T>
+             */
+            static function (bool $throwOnException): Closure {
+                return
+                    /**
+                     * @psalm-param Iterator<TKey, T> $iterator
+                     *
+                     * @psalm-return Closure(Iterator<TKey, T>): Generator<TKey, T>
+                     */
+                    static function (Iterator $iterator) use ($throwOnException): Closure {
+                        // As keys can be of any type in this library.We cannot use
+                        // iterator_to_array() because it expect keys to be int|string.
+                        try {
+                            foreach ($iterator as $key => $value) {
+                            }
+                        } catch (Throwable $e) {
+                            if (true === $throwOnException) {
+                                throw $e;
+                            }
```

Comment:
what's the use case here for swallowing the exception if `throwOnException` is _false_? Just trying to understand what this could be used for.

Reply:
This is something that came to my mind. We could imagine to ignore them or not. I don't like it so much, I'm still thinking about doing that in a different way. Do you have a better suggestion?

Reply:
After thinking more about it, I removed it.

Reply:
Ignoring exceptions I think would be a bit unexpected. A different use case I can think of is if you passed a closure which you wanted to execute in the try-catch - let's say you wanted to add some logging if an exception occurs and then re-throw it. However, I feel like that's going more towards a method that you would want to use specifically for exception handling, so probably best to exclude it from this 👍